src/cache/gcs.rs

```diff
@@ -36,6 +37,15 @@ impl RWMode {
     }
 }
 
+impl From<config::CacheRWMode> for RWMode {
+    fn from(value: config::CacheRWMode) -> Self {
+        match value {
+            config::CacheRWMode::ReadOnly => RWMode::ReadOnly,
+            config::CacheRWMode::ReadWrite => RWMode::ReadWrite,
+        }
+    }
+}
+
```

Comment:
There are a few different types like this that have the same purpose. Should they be consolidated?

Reply:
I've merged two of them together, although there's still `CacheModeConfig` and `CacheMode`, which do effectively the same thing.